src/backend/access/appendonly/aosegfiles.c

```diff
@@ -1527,25 +1527,25 @@ get_ao_compression_ratio(PG_FUNCTION_ARGS)
 	Relation	parentrel;
 	float8		result = -1.0;
 
+	/* open the parent (main) relation */
+	parentrel = table_open(relid, AccessShareLock);
```

Comment:
use `get_rel_relkind` is simple.

Reply:
From origin commit: https://github.com/apache/cloudberry/pull/845/commits/34582c9f8d0ca44845f2da46964444b7064cb0a1